### PR TITLE
User should be able to update his user data (Profile)

### PR DESCRIPTION
**Pull request by nadezhdakharchuk**
# Summary

#### Screenshots in case of UI changes
screencast 2020-08-04 12-22-02

# Test plan

# Review notes

#### While reviewing pull-request (especially when it's your pull-request), please make sure that:

- you named PR close to issue name and linked that issue
- you understand what problem is solved by PR and how is it solved
- new tests are in place, no redundant tests
- required ENV variables added and described in `.env.example` and added to Heroku
- associated Heroku review app works correctly with introduced changes

### CODE REVIEW

File: components/organisms/ProfileForm/ProfileForm.test.js [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    const expectedValues = {
+      ...expectedProfile,
+      password: '',
+      currentPassword: '',
+      [expectedButtonText]: expectedButtonText,
+    };
+    const mockUpdateUser = jest.fn(() => Promise.resolve());
+    actions.useUpdateUser = jest.fn(() => [mockUpdateUser]);
+
+    const { getByText } = render(renderWithTheme(<ProfileForm profile={expectedProfile} />));
```
Comment (RenasSitdikov) [inline components/organisms/ProfileForm/ProfileForm.test.js:43]: Can we use `MockedProvider` from `@apollo/react-testing` as in Premiumparking? We can test with mutations in this case

File: components/organisms/ProfileForm/ProfileFormContent.js [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+      initialValue: '',
+      validationSchema: Yup.string(),
+    },
+    {
+      type: 'password',
+      name: 'currentPassword',
+      title: 'Confirm Password',
+      placeholder: 'Confirm Password',
+      initialValue: '',
+      validationSchema: Yup.string().oneOf([Yup.ref('password'), null], 'Passwords must match'),
```
Comment (Nikita-Polyakov) [inline components/organisms/ProfileForm/ProfileFormContent.js:59]: This field should have title and placeholder 'Current Password' and shouldn't match a new password

File: components/organisms/ProfileForm/ProfileFormContent.js [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+        .email('The email must be valid!!')
+        .required('This field is required'),
+    },
+    {
+      type: 'password',
+      name: 'password',
+      title: 'New Password',
+      placeholder: 'New Password',
+      initialValue: '',
+      validationSchema: Yup.string(),
```
Comment (Nikita-Polyakov) [inline components/organisms/ProfileForm/ProfileFormContent.js:51]: I think validation rule should be like: 'if you filled New Password, Current Password field should be filled too'

File: components/organisms/ProfileForm/ProfileForm.test.js (hunk 1 above)
Comment (nadezhdakharchuk) [reply components/organisms/ProfileForm/ProfileForm.test.js:43]: We know nothing about apollo in this component. I think we can use `MockedProvider` in the Profile page test

File: __tests__/pages/profile/index.test.js [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+    jest.resetModules();
+    cleanup();
+  });
+
+  test('should render correctly', async () => {
+    // Arrange
+    const mockUseCurrentUserData = jest.fn(() => ({
+      loading: undefined,
+      error: undefined,
+      user: { id: '1', email: 'user@mail.ru', firrstName: 'First Name' },
```
Comment (Nikita-Polyakov) [inline __tests__/pages/profile/index.test.js:29]: a typo here `firrstName` -> `firstName`

File: components/atoms/ErrorMessage/ErrorMessage.js [hunk 5]
```diff
@@ -1,7 +1,9 @@
 import React from 'react';
 
-const ErrorMessage = ({ children }) => (
-  <div>{Array.isArray(children) ? children.map((error, index) => <p key={index}>{error}</p>) : <p>{children}</p>}</div>
+const ErrorMessage = ({ children, testId }) => (
```
Comment (Nikita-Polyakov) [inline components/atoms/ErrorMessage/ErrorMessage.js:3]: maybe we can use a rest spread pattern here?

```
const ErrorMessage = ({ children, ...attributes }) => (
  <div {...attributes}>
    {Array.isArray(children) ? children.map((error, index) => <p key={index}>{error}</p>) : <p>{children}</p>}
  </div>
);
```

**Pull request merged**